pkg/kn/commands/config/channel_type_mapping.go

```diff
@@ -0,0 +1,175 @@
+// Copyright © 2022 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configCmd
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+	"knative.dev/client/pkg/kn/commands"
+	"knative.dev/client/pkg/kn/config"
+)
+
+var defaultChannelTypeUpdater = viperUpdater(viper.GetViper(), config.KeyChannelTypeMappings)
+
+const (
+	ChannelTypeAliasNotFoundErrorMsg = "Channel type alias %q was not found in the configuration file"
+	ChannelTypeAliasExistsErrorMsg   = "Channel type alias %q already exists in the configuration file"
+)
+
+// NewConfigChannelTypeMappingCommand represents 'kn config channel-type-mapping' command
+func NewConfigChannelTypeMappingCommand(p *commands.KnParams) *cobra.Command {
+	channelTypeMappingCmd := &cobra.Command{
+		Use:   "channel-type-mapping",
+		Short: "Channel type mappings configuration settings",
+	}
+	channelTypeMappingCmd.AddCommand(newChannelTypeDeleteCommand(p))
+	channelTypeMappingCmd.AddCommand(newChannelTypeAddCommand(p))
+	channelTypeMappingCmd.AddCommand(newChannelTypeUpdateCommand(p))
+	channelTypeMappingCmd.AddCommand(newChannelTypeListCommand(p))
+	return channelTypeMappingCmd
+}
+
+func addChannelTypeMappingFlags(flagSet *pflag.FlagSet, channelType *config.ChannelTypeMapping) {
+	flagSet.StringVar(&channelType.Kind, "kind", "", "The kind of Channel, such as InMemoryChannel or KafkaChannel, based on the default ConfigMap")
+	flagSet.StringVar(&channelType.Group, "group", "", "The API group of the Kubernetes resource")
+	flagSet.StringVar(&channelType.Version, "version", "", "The version of the Kubernetes resource")
+}
+
+func newChannelTypeUpdateCommand(_ *commands.KnParams) *cobra.Command {
+	channelType := config.ChannelTypeMapping{}
+	updateMapping := func(cmd *cobra.Command, args []string) ([]interface{}, error) {
```

Comment:
I wonder if for consistency sake whether these func definition should be placed outside or consistent order as the other commands in the client?

Reply:
+1 to keep the consistency with other commands. Ideally channel and sink mappings would live in their own sub-packages/directories and commands split into files.

```
./config/channel/<per_crud_cmds>.go
./config/sink/<per_crud_cmd>.go
```

Reply:
@dsimansk & @maximilien DONE